maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/VersionSchemeProvider.java

```diff
@@ -0,0 +1,49 @@
+package org.apache.maven.repository.internal;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.eclipse.aether.util.version.GenericVersionScheme;
+import org.eclipse.aether.version.VersionScheme;
+
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+/**
+ * Provider of {@link org.eclipse.aether.util.version.GenericVersionScheme}.
+ */
+@Named
+@Singleton
+public class VersionSchemeProvider
```

Comment:
Why do we need this provider? Where is it used?

Reply:
Need to review this and refresh my memory. But the idea was to have some sort of indirection, if we want to introduce other kind than "generic" version support... Consider this PR as draft pls